### ISSUE TITLE
Accessibility: `OptionSetField` and `CheckboxSetField` options missing `role="option"`

### ISSUE THREAD

**Issue by GuySartorelli** [silverstripe/silverstripe-admin#1964]
[ARIA 1.2](https://www.w3.org/TR/wai-aria-1.2/) [WCAG 2.2 A 1.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html) Elements with role=listbox must contain or own an element with role=option and must not contain elements with other roles

The parent `div` for checkboxes in a `CheckboxSetField` and radio buttons in a `OptionSetField` have a `role="listbox"` attribute.
The individual checkboxes and radio buttons are _supposed_ to have a `role="option"` attribute but that is missing.
The PHP code is setting those attributes, so something is likely wrong with the templates.

## Acceptance criteria

- [ ] Checkboxes in a  `CheckboxSetField` have a `role="option"` attribute
- [ ] Radio buttons in a  `OptionSetField` have a `role="option"` attribute
- [ ] Check both entwine and react forms

## PRs
- https://github.com/silverstripe/silverstripe-admin/pull/1992
- https://github.com/silverstripe/silverstripe-framework/pull/11847

**Comment by GuySartorelli** [comment]
PRs merged
